libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -44,7 +49,7 @@ internal class MapboxDirectionsSession(
         @RoutesExtra.RoutesUpdateReason routesUpdateReason: String
     ) {
         this.initialLegIndex = initialLegIndex
```

Comment:
out of scope of the PR, but why do we set `initialLegIndex` before the `routes`. It looks like the routes might not be set, but `initialLegIndex` always be 🤔

Reply:
cc @LukasPaczos @VysotskiVadim

Reply:
Interesting. It doesn't seem like a big deal, because we don't update routes only if old and new routes are both empty. From the other side I can't imagine a case when you need to update the initialLegIndex for an empty route, so it doesn't make sense for me either 🙂 
I vote for moving index update so that it happens after the check to avoid confusion.

Reply:
let's cut a ticket, it doesn't look like a big deal, but still, need to investigate what components depend on it

Reply:
BTW, git blame says it was done in 65a66fd by @Zayankovsky . Can you please confirm that index is updated before conditional routes update accidentally, i.e. without any specific purpose?

Reply:
You shouldn't move leg index update so that it happens after the check, because in `MapboxNavigation.internalSetNavigationRoutes` this leg index is passed to trip session and hence to nav-native. Which means you should always update the index here for consistency.